components/display/nextion.rst

```diff
@@ -241,35 +241,16 @@ In either case, this automation can be helpful to update on-screen controls for
 If you fully own your Nextoin HMI design and follow the best practice of setting the components' vscope to global in the Nextion Editor, you'll probably never need this trigger.
 However, if this is not the case and all / some of your UI components have local visibility scope, ``on_page`` will be your remedy. Here you can initiate updates of the relevant components.
 
-Before actually updating components, you need to understand which page Nextion was switched to. For ESP initiated page changes ``x`` argument will contain a string
-(``std::string`` to be precise) with the new page name. In case of Nextion initiated change, page names are not available and ``x`` will be a page id as string.
-If the same page can be enabled both from ESP and from Nextion, be prepared to parse both the name and the id of that page.
-
-.. code-block:: yaml
-
-    on_page:
-      then:
-        lambda: |-
-          // x contains an std::string with a page name or page id.
-          //   Both can be used for goto_page() function
-          int page_id = std::atoi(x.c_str());
-          if (page_id == 0 && x != "0") {
-            // manually convert page names to ids
-            if      (x == "main_page")    page_id = 0x01;
-            else if (x == "wifi_qr_page") page_id = 0x02;
-            ...
-          }
-
+Before actually updating components, you need to understand which page Nextion was switched to. ``x`` argument will contain a page id ingeger.
```

Comment:
integer

Reply:
Thank you!
Note to self: always use a spell checker, understand why my spell checker treats ingeger as a valid word (just tried again, still no complains)